internal/jobservice/repository/job_service_in_memory.go

```diff
@@ -0,0 +1,156 @@
+package repository
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	js "github.com/G-Research/armada/pkg/api/jobservice"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type JobStatus struct {
+	jobMap        map[string]*JobTable
+	jobLock       sync.RWMutex
+	subscribeMap  map[string]*SubscribeTable
+	subscribeLock sync.RWMutex
+}
+
+func NewJobStatus(jobMap map[string]*JobTable, subscribeMap map[string]*SubscribeTable) *JobStatus {
+	return &JobStatus{jobMap: jobMap, subscribeMap: subscribeMap}
+}
+
+type InMemoryJobServiceRepository struct {
+	jobStatus        *JobStatus
+	jobServiceConfig *configuration.JobServiceConfiguration
+}
+
+func NewInMemoryJobServiceRepository(jobMap *JobStatus, config *configuration.JobServiceConfiguration) *InMemoryJobServiceRepository {
+	return &InMemoryJobServiceRepository{jobStatus: jobMap, jobServiceConfig: config}
+}
+
+func (inMem *InMemoryJobServiceRepository) GetJobStatus(jobId string) (*js.JobServiceResponse, error) {
+	inMem.jobStatus.jobLock.RLock()
```

Comment:
You could consider using sync/atomic.Value to avoid having to do the locking yourself. I'm not 100% sure it's a perfect fit though.  

https://pkg.go.dev/sync/atomic#Value

Reply:
I think I need this bcuz maps by default aren’t threadsafe.  I think go throws a panic if they are accessed concurrently.

Reply:
I am pretty sure this is exactly the case that the `sync/atomic` package is designed for. I've only used it for inc/dec integers across threads though, not higher order objects.

Either way, I wanted to make sure you were aware this existed in the stdlib; there may be good reasons to not use it (including but not limited to 'this code is already written and I didn't :D).

Reply:
I think you're right, this is a bad fit on further analysis. `sync/atomic` only validates that a write can happen before the read; but it does nothing to ensure multiple writers are not updating simultaneously. It seems like if anything, you'd want to use it *in addition to* more locking.

Reply:
Yea there is a sync map that I could potentially use but it seemed more complicated than I needed.